From ac8fe854642271b6585ab47c4749e6cb1c5aab04 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 18:58:33 -0400
Subject: [idle] intel_idle: do C1E promotion disable quirk for hotplugged CPUs

Message-id: <1405364315-27440-6-git-send-email-prarit@redhat.com>
Patchwork-id: 85798
O-Subject: [RHEL7.1 PATCH BZ 1119374 5/7] intel_idle: do C1E promotion disable quirk for hotplugged CPUs
Bugzilla: 1119374
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119374

commit dbf87ab89fbd14b723b7282de635bc70f4996342
Author: Bartlomiej Zolnierkiewicz <b.zolnierkie@samsung.com>
Date:   Fri Dec 20 19:47:28 2013 +0100

    intel_idle: do C1E promotion disable quirk for hotplugged CPUs

    If the system is booted with some CPUs offline C1E promotion disable quirk
    won't be applied because on_each_cpu() in intel_idle_cpuidle_driver_init()
    operates only on online CPUs. Fix it by adding the C1E promotion disable
    handling to intel_idle_cpu_init() (which is also called during CPU_ONLINE
    operation).

    Signed-off-by: Bartlomiej Zolnierkiewicz <b.zolnierkie@samsung.com>
    Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
    Cc: Len Brown <lenb@kernel.org>
    Reviewed-by: Daniel Lezcano <daniel.lezcano@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 9901a62..a14bbd8 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -626,6 +626,9 @@ static int intel_idle_cpu_init(int cpu)
  if (icpu->auto_demotion_disable_flags)
   smp_call_function_single(cpu, auto_demotion_disable, NULL, 1);
 
+ if (icpu->disable_promotion_to_c1e)
+  smp_call_function_single(cpu, c1e_promotion_disable, NULL, 1);
+
  return 0;
 }
 
-- 
1.7.1